tests/test_s3.py

```diff
@@ -1543,6 +1543,64 @@ def test_has_statement_policy(self):
         resources = p.run()
         self.assertEqual(len(resources), 1)
 
+    def test_bucket_replication_policy_remove(self):
+        self.patch(s3.S3, "executor_factory", MainThreadExecutor)
+        self.patch(
+            s3.SetBucketReplicationConfig, "executor_factory", MainThreadExecutor
+        )
+        self.patch(
+            s3, "S3_AUGMENT_TABLE", [("get_bucket_policy", "Policy", None, "Policy")]
+        )
+        session_factory = self.replay_flight_data("test_s3_replication_policy_remove")
+        bname = "repela"
+        p = self.load_policy(
+            {
+                "name": "s3-has-replica-policy",
+                "resource": "s3",
+                "filters": [
+                    {"Name": bname}
```

Comment:
this should filter for pre-condition, that the bucket replication is present.

Reply:
Agreed! Resolved in 473e39223aa92715f18e5ce6a3a46e82d29759aa